### PR TITLE
Added powsimp in _combine_inverse

### PR DESCRIPTION
**Pull request by ritesh99rakesh**
#### References to other Issues or PRs

Please refer to discussion on PR  https://github.com/sympy/sympy/pull/16848#issuecomment-494740582

#### Brief description of what is fixed or changed
Before
```
>>> x, a = symbols('x a')
>>> Mul._combine_inverse(x*x**a, x**a)
x
>>> Mul._combine_inverse(x*x**(a+1), x**(a+1))
x*x**(-a - 1)*x**(a + 1)
```
After
```
>>> x, a = symbols('x a')
>>> Mul._combine_inverse(x*x**a, x**a)
x
>>> Mul._combine_inverse(x*x**(a+1), x**(a+1))
x
>>> Mul._combine_inverse(x*x**(y+1), x**(y+1))
x
```
Modified `sympy/core/mul.py` and added test to `sympy/core/tests/test_match.py`
Changes suggested by @jksuom 
#### Release Notes

- core
    - Corrected `_combine_inverse` under `sympy/core/test_match.py`

### CODE REVIEW

**Comment by ritesh99rakesh** [comment]
@jksuom @sidhantnagpal please review this.

**Comment by jksuom** [comment]
> exponents of a common base always combine

This is what I also expected but it looks like there are some exceptions that made the match fail. I am not sure if they are essential. Here is one example that I encountered when trying to change `Mul.flatten`.
https://github.com/sympy/sympy/blob/bb92e8b506bfe861328b51c42e38532031524281/sympy/core/tests/test_arit.py#L1413-L1415

In any case, it should not be necessary to modify `_flatten`. This direct treatment in `_combine_inverse` goes straight to the point and looks fine to me.

File: sympy/core/mul.py [hunk 1]
```diff
@@ -1044,18 +1044,17 @@ def check(l, r):
...
+            a = lhs.as_powers_dict()
+            b = rhs.as_powers_dict()
+            blen = len(b)
+            for bi in tuple(b.keys()):
+                if bi in a:
+                    a[bi] -= b.pop(bi)
+            if len(b) != blen:
+                # do not recompute powers if exponent is 0
+                lhs = Mul(*[b**e for b, e in a.items() if e])
+                rhs = Mul(*[b**e for b, e in b.items() if e])
```
Comment (jksuom) [inline sympy/core/mul.py]: `b` has double role.

**Comment by jksuom** [comment]
It looks like `as_powers_dict` does not work as expected:
```
>>> lhs = -oo*I*x
>>> rhs = oo*I
>>> lhs.as_powers_dict()
defaultdict(<type 'int'>, {-oo: 1, x: 1, -1: 1/2})
>>> rhs.as_powers_dict()
defaultdict(<type 'int'>, {-1: 1/2, oo: 1})
```
(On the other hand, `-oo*I*x/oo*I` does not make much sense mathematically.)

**Comment by smichr** [comment]
> does not make much sense mathematically

This method was designed to handle such cases. But it looks like `-oo` needs its `as_powers_dict` method added or modified.

**Comment by smichr** [comment]
Two test failures...not sure why:
```
______________ sympy/series/tests/test_gruntz.py:test_issue_7096 _______________
Traceback (most recent call last):
  File "utilities/runtests.py", line 1308, in _timeout
    function()
  File "series/tests/test_gruntz.py", line 469, in test_issue_7096
    assert gruntz(x**-pi, x, 0, dir='-') == oo*sign((-1)**(-pi))
AssertionError
________________________________________________________________________________
________________ sympy/series/tests/test_limits.py:test_basic1 _________________
Traceback (most recent call last):
  File "utilities/runtests.py", line 1308, in _timeout
    function()
  File "series/tests/test_limits.py", line 77, in test_basic1
    assert limit(x**-pi, x, 0, dir='-') == oo*sign((-1)**(-pi))
AssertionError
```

**Pull request closed without merging**